react/src/views/ScanBox.tsx

```diff
@@ -1,59 +1,110 @@
 import React, { useState } from "react";
+import { useLazyQuery } from "@apollo/client";
 import QrReader from "react-qr-reader";
 import { Link } from "react-router-dom";
 import { Button, Icon, Header } from "semantic-ui-react";
+import { QR_EXISTS, QR_BOX_EXISTS } from "../utils/queries";
+import { Redirect } from "react-router";
 
 function ScanBox() {
-  const [data, setData] = useState("");
+  const [qr, setQR] = useState("");
+  const [qrExists, setQrExists] = useState(null);
+  const [qrBoxExists, setQrBoxExists] = useState(null);
   const [qrError, setQrError] = useState("");
 
+  const [getQrExistsQuery] = useLazyQuery(QR_EXISTS, {
+    onCompleted: (data) => {
+      setQrExists(data);
+    },
+  });
+
+  const [getQrBoxExistsQuery] = useLazyQuery(QR_BOX_EXISTS, {
+    onCompleted: (data) => {
+      setQrBoxExists(data);
+    },
+  });
+
+  const retrieveBox = (code) => {
+    if (code) {
+      const myQR = code.split("barcode=")[1];
+      setQR(myQR);
+      try {
+        getQrExistsQuery({
+          variables: {
+            qrCode: String(myQR),
+          },
+        });
+        getQrBoxExistsQuery({
```

Comment:
If the QR code does not exist this part wouldn't even be called?

Reply:
Good call, I just changed the workflow to better take advantage of GraphQL and fetch both pieces of info in one query